utils/agreement-utils/src/manifest.rs

```diff
@@ -0,0 +1,374 @@
+use std::ops::Not;
```

Comment:
I disagree, but I'll extract it since it will take less time than reaching a consensus.

- manifests exist only within the context of agreements
- this code is for (de)serialization and signature verification only
- manifest support is feature-enabled
- `agreement-utils` already aren't protocol agnostic, since `typed_props.rs` contain some hardware / runtime information